tests/rules/test_underscore_output.py

```diff
@@ -0,0 +1,179 @@
+    workflow = """\
+---
+name: Test Correct Workflow
+on:
+  workflow_dispatch:
+    outputs:
+      registry:
+        value: 'Test Value'
+      some_registry:
+        value: 'Test Value'
+  workflow_call:
+    outputs:
+      registry:
+        value: 'Test Value'
+      some_registry:
+        value: 'Test Value'
+  push: {}
+
+jobs:
+  job-key:
+    name: Test Correct Job
+    runs-on: ubuntu-22.04
+    outputs:
+      test_key_job: ${{ steps.test_output_1.outputs.test_key }}
+    steps:
+      - name: Test output in one-line run step
+        id: test_output_1
+        run: echo "test_key_1=Test-Value1" >> $GITHUB_OUTPUT
+
+      - name: Test output in multi-line run step
+        id: test_output_2
+        run: |
+          echo
+          fake-command=Test-Value2
+          echo "test_key_2=$REF" >> $GITHUB_OUTPUT
+          echo "deployed_ref=$DEPLOYED_REF" >> $GITHUB_OUTPUT
+
+      - name: Test step with one-line run and no Output
+        id: test_output_3
+        run: echo "test_key_3"
+
+      - name: Test step with multi-line run and no Output
+        id: test_output_4
+        run: |
+          echo
+          fake-command=Test-Value4
+          echo "test_key_4"
+          echo "deployed_ref"
+
+      - name: Test step with no run
+        id: test_output_5
+        uses: actions/checkout@v2
+        with:
+          ref: ${{ github.ref }}
+          fetch-depth: 0
+"""
+    return WorkflowBuilder.build(workflow=yaml.load(workflow), from_file=False)
+
+
+@pytest.fixture(name="incorrect_workflow")
+def fixture_incorrect_workflow():
+    workflow = """\
+---
+name: Test Incorrect Workflow
+on:
+  workflow_dispatch:
+    outputs:
+      registry-1:
+        value: 'Test Value'
+      some_registry-1:
+        value: 'Test Value'
+  workflow_call:
+    outputs:
+      registry-2:
+        value: 'Test Value'
+  push: {}
+
+jobs:
+  job-key:
+    name: Test Incorrect Job
+    runs-on: ubuntu-22.04
+    outputs:
+      test-key-1: ${{ steps.test_output_1.outputs.test_key }}
+    steps:
+      - name: Test output in one-line run step
+        id: test_output_1
+        run: echo "test-key-1=Test-Value1" >> $GITHUB_OUTPUT
+
+      - name: Test output in multi-line run step
+        id: test_output_2
+        run: |
+          echo
+          fake-command
+          echo "test-key-2=$REF" >> $GITHUB_OUTPUT
+          echo "deployed-ref=$DEPLOYED_REF" >> $GITHUB_OUTPUT
+
+      - name: Test step with one-line run and no Output
+        id: test_output_3
+        run: echo "test-key-3"
+
+      - name: Test step with multi-line run and no Output
+        id: test_output_4
+        run: |
+          echo
+          fake-command=Test-Value4
+          echo "test-key-4"
+          echo "deployed-ref"
+"""
+    return WorkflowBuilder.build(workflow=yaml.load(workflow), from_file=False)
+
+
+@pytest.fixture(name="rule")
+def fixture_rule():
+    return RuleUnderscoreOutputs()
+
+
+def test_rule_on_correct_workflow(rule, correct_workflow):
```

Comment:
I'd recommend adding some checks against the expected messaging in some of the correct and incorrect tests.

Reply:
Do you mean `incorrect` tests, because we only return `True` & `""` when the test is correct

Reply:
Sorry! Yeah, I meant the `incorrect` tests.